plugins/modules/ip2locationio_facts.py

```diff
@@ -0,0 +1,160 @@
+#!/usr/bin/python
+
+# Copyright (c) 2025, IP2Location <support@ip2location.com>
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import annotations
+
+DOCUMENTATION = r"""
+module: ip2locationio_facts
+short_description: Retrieve IP geolocation facts of a host's IP address
+version_added: 13.1.0
+description:
+  - Gather IP geolocation facts of a host's IP address using the keyless U(api.ip2location.io) API.
+author: "IP2Location (@ip2location)"
+extends_documentation_fragment:
+  - community.general.attributes
+  - community.general.attributes.facts
+  - community.general.attributes.facts_module
+options:
+  ip:
+    description:
+      - IP address to retrieve geolocation facts for.
+    type: str
```

Comment:
If the IP address is configurable, this shouldn't be a `_facts` module, but an `_info` module.